elasticsearch/run.sh

```diff
@@ -57,17 +66,76 @@ else
 	exit 1
 fi
 
-TIMES=300
-function waitForES() {
-  for (( i=1; i<=$TIMES; i++ )); do
+# Wait for Elasticsearch port to be opened. Fail on timeout or if response from Elasticsearch is unexpected.
+wait_for_port_open() {
+	rm -f $LOG_FILE
     # test for ES to be up first and that our SG index has been created
-		result=$(curl --cacert $secret_dir/admin-ca --cert $secret_dir/admin-cert --key $secret_dir/admin-key -s -w "%{http_code}" -XGET "https://localhost:9200/.searchguard.${HOSTNAME}" -o /dev/null) ||:
-    [[ $result -eq 200 ]] && return 0
-    sleep 1
-  done
+	echo -n "Checking if Elasticsearch is ready on $ES_REST_BASEURL "
+	while ! response_code=$(curl -s \
+			--cacert $secret_dir/admin-ca \
+			--cert $secret_dir/admin-cert \
+			--key  $secret_dir/admin-key \
+			--max-time $max_time \
+			-o $LOG_FILE -w '%{response_code}' \
+			"$ES_REST_BASEURL/.searchguard.${HOSTNAME}") || test $response_code != "200"
+	do
+		echo -n "."
+		sleep $RETRY_INTERVAL
+		(( retry -= 1 ))
+		if (( retry == 0 )) ; then
+			timeouted=true
+            break
+		fi
+	done
+
+	if [ $timeouted = true ] ; then
+		echo -n "[timeout] "
+    else
+        rm -f $LOG_FILE
+        return 0
+	fi
+	echo "failed"
+	cat $LOG_FILE
+    rm -f $LOG_FILE
+	exit 1
+}
+
+verify_or_add_index_templates() {
+	wait_for_port_open
+	# Uncomment this if you want to wait for cluster becoming more stable before index template being pushed in.
+	# Give up on timeout and continue...
+	# curl -v -s -X GET \
+	#	--cacert $secret_dir/admin-ca \
+	#	--cert $secret_dir/admin-cert \
+	#	--key  $secret_dir/admin-key \
+	#	"$ES_REST_BASEURL/_cluster/health?wait_for_status=yellow&timeout=${max_time}s"
 
-  echo "Was not able to connect to Elasticearch at localhost:9200 within $TIMES attempts"
+	shopt -s failglob
+	for template_file in /usr/share/elasticsearch/index_templates/*.json
+	do
+		template=`basename $template_file`
+		# Check if index template already exists
+		response_code=$(curl -s -X HEAD \
+			--cacert $secret_dir/admin-ca \
+			--cert $secret_dir/admin-cert \
+			--key  $secret_dir/admin-key \
+			-w '%{response_code}' \
+			$ES_REST_BASEURL/_template/$template)
+		if [ $response_code == "200" ]; then
+			echo "Index template '$template' already present in ES cluster"
+		else
+			echo "Create index template '$template'"
```

Comment:
I suppose, but if the code is not 200 (404 or 5xx or whatever) it will attempt to add the template, which will then fail, assuming the previous condition was not 404, or perhaps it will fail anyway if there is some other problem, and that will be reported.